pnpbridge/src/adapters/src/mqtt_pnp/mqtt_pnp.cpp

```diff
@@ -136,82 +134,23 @@ MqttPnp_CreatePnpInterface(
 
     context->s_ProtocolHandler->Initialize(&context->s_ConnectionManager, adapterConfig);
 
-    result = DigitalTwin_InterfaceClient_Create(InterfaceId,
-                                               ComponentName,
-                                               nullptr,
-                                               context, 
-                                               &pnpInterfaceClient);
-    if (DIGITALTWIN_CLIENT_OK != result)
-    {
-        LogError("mqtt-pnp: Error registering pnp interface %s", InterfaceId);
-        result = DIGITALTWIN_CLIENT_ERROR;
-        goto exit;
-    }
-
-    result = DigitalTwin_InterfaceClient_SetPropertiesUpdatedCallback(pnpInterfaceClient,
-        [](
-            const DIGITALTWIN_CLIENT_PROPERTY_UPDATE*   /*dtClientPropertyUpdate*/,
-            void*                                       /*userContextCallback*/
-        )
-        {
-            return;
-        },
-        (void*)context
-    );
-
-    if (DIGITALTWIN_CLIENT_OK != result)
-    {
-        LogError("mqtt-pnp: Error binding property update callback for %s", InterfaceId);
-        result = DIGITALTWIN_CLIENT_ERROR;
-        goto exit;
-    }
-
-    result = DigitalTwin_InterfaceClient_SetCommandsCallback(pnpInterfaceClient,
-        [](
-            const DIGITALTWIN_CLIENT_COMMAND_REQUEST*   dtClientCommandContext,
-            DIGITALTWIN_CLIENT_COMMAND_RESPONSE*        dtClientCommandResponseContext,
-            void*                                       userContextCallback
-        )
-        {
-            static_cast<MqttPnpInstance*>(userContextCallback)->s_ProtocolHandler->OnPnpMethodCall(dtClientCommandContext, dtClientCommandResponseContext);
-        },
-        (void*)context
-    );
-
-    if (DIGITALTWIN_CLIENT_OK != result)
-    {
-        LogError("mqtt-pnp: Error binding commands callback for %s", InterfaceId);
-        result = DIGITALTWIN_CLIENT_ERROR;
-        goto exit;
-    }
-
-    *PnpInterfaceClient = pnpInterfaceClient;
-    context->s_ProtocolHandler->AssignDigitalTwin(pnpInterfaceClient);
-    PnpInterfaceHandleSetContext(BridgeInterfaceHandle, context);
+    PnpComponentHandleSetContext(PnpComponentHandle, context);
+    PnpComponentHandleSetPropertyUpdateCallback(PnpComponentHandle, JsonRpcProtocolHandler::OnPnpPropertyCallback);
```

Comment:
>JsonRpcProtocolHandler [](start = 68, length = 22)

as noted in JsonRpcProtocolHandler - these dispatch functions can be general if we make them part of MqttProtocolHandler, then they'll call into the actual s_ProtocolHandler in the context